Sources/Testing/Expectations/Expectation.swift

```diff
@@ -62,3 +62,65 @@ public struct ExpectationFailedError: Error {
   /// The expectation that failed.
   public var expectation: Expectation
 }
+
+// MARK: - Snapshot
+
+extension Expectation {
+  /// A serializable type describing an expectation that has been evaluated.
+  @_spi(ExperimentalSnapshotting)
+  public struct Snapshot: Sendable, Codable {
+    /// The source code of the expression evaluated by this expectation, if
+    /// available at compile time.
+    public var sourceCode: String?
+
+    /// A description of the error mismatch that occurred, if any.
+    ///
+    /// If this expectation passed, the value of this property is `nil` because no
+    /// error mismatch occurred.
+    @_spi(ExperimentalEventHandling)
```

Comment:
I'm not sure what to do about the `@_spi` annotations on these stored properties given that this is a snapshot we'd want to immediately encode to JSON or something. @stmontgomery What do you think?

Reply:
I'm not too concerned by this, and it matches the `@_spi` declarations of the original properties.